src/components/PositionedOverlay/tests/PositionedOverlay.test.tsx

```diff
@@ -243,6 +243,13 @@ describe('<PositionedOverlay />', () => {
       ).toBe(false);
     });
   });
+
+  describe('mutationObserveConfig', () => {
```

Comment:
Anyone know of a good example I can reference for testing a mutation observer?

Reply:
I don't think we have any example of testing mutation observer. But we might be able to override the prototype & spy on it?

```tsx
  // ⬇️  untested "pseudo" code
  let mutationObserverObserveSpy: jest.SpyInstance;

  beforeEach(() => {
    mutationObserverObserveSpy = jest.spyOn(MutationObserver.prototype, 'observe');
  });

  afterEach(() => {
    mutationObserverObserveSpy.mockRestore();
  });
  
  it(...) {
  ...
  expect(mutationObserverObserveSpy).toHaveBeenCalledWith(expect.any(Function), CONFIG)
  } 
  
  ```

Reply:
Nice - thanks @AndrewMusgrave .

I needed to use the DOM node for the first argument, but other than that, seems like this worked as intended 👍 

@kyledurand / @AndrewMusgrave mind giving another review?

Another thing to consider... and @clauderic can chime in on this as well, but perhaps we want listening for `attributes` changes to be the default behaviour? Maybe we don't even need this new `mutationObserveConfig` prop? What do you both think?

Reply:
@kyledurand / @AndrewMusgrave what are your thoughts on @clauderic comments? Should I go ahead with the PR as is, or should we always observe `attribute` changes?

Depending on the answer, would we still want to support the new `mutationObserveConfig` prop? It does allow for more flexibility, but just how valuable of a feature that is for our consumers I don't know.

Reply:
I agree with @clauderic. Polaris components should work out of the box. If adding `attributes` to the config fixes a bug, you shouldn't have to opt into the fix 😄  I don't suspect we'll face a performance issue, but if we do. We can always evaluate making an API change then!